libs/stats/odc/stats/_fc_percentiles.py

```diff
@@ -0,0 +1,110 @@
+"""
+Fractional Cover Percentiles
+"""
+from typing import Optional, Tuple
+from itertools import product
+import dask.array as da
+import xarray as xr
+from odc.stats.model import Task
+from odc.algo.io import load_with_native_transform
+from odc.algo import keep_good_only
+from odc.algo._percentile import xr_percentile
+from odc.algo._masking import _nodata_fuser, _or_fuser
+from .model import StatsPluginInterface
+from . import _plugins
+
+
+class StatsFCP(StatsPluginInterface):
+    
+    NAME = "ga_fc_percentiles"
+    SHORT_NAME = NAME
+    VERSION = "0.0.1"
+    PRODUCT_FAMILY = "fc_percentiles"
+
+    def __init__(
+        self,
+        resampling: str = "bilinear",
+    ):
+        self.resampling = resampling
+
+    @property
+    def measurements(self) -> Tuple[str, ...]:
+        _measurments = [f"{b}_pc_{p}" for b, p in product(["pv", "bs", "npv", "ue"], ["10", "50", "90"])]
+        _measurments.append("is_ever_wet")
+        return _measurments
+
+    @staticmethod
+    def _native_tr(xx):
+        """
+        Loads in the data in the native projection. It performs the following:
+
+        1. Loads all the fc and WOfS bands
+        2. Set high slope terrain flag to 0
+        3. Extracts the clear dry flags from WOfS
+        4. Drops the WOfS band
+        5. Masks out all pixels that are not clear and dry to a nodata value of 255
+        6. Discards the clear dry flags
+        """
+
+        # set terrain flag to zero
+        water = da.bitwise_and(xx["water"], 0b11101111)
+        xx = xx.drop_vars(["water"])
+
+        # use the dry flag to indicate good measurements
+        dry = water == 0
+        xx = keep_good_only(xx, dry, nodata=255)
+        return xx
+
+    @staticmethod
+    def _native_tr_wet(xx):
+        """
+        Loads in the data in the native projection. It performs the following:
+
+        1. Loads the WOfS band
+        2. Set high slope terrain flag to 0
+        3. Extracts clear wet flags from WOfS
+        """
+
+        # set terrain flag to zero
+        water = da.bitwise_and(xx["water"], 0b11101111)
+        wet = water == 128
+        return wet
+
+    def input_data(self, task: Task) -> xr.Dataset:
+
+        chunks = {"y": -1, "x": -1}
+
+        xx = load_with_native_transform(
+            task.datasets,
+            bands=["water", "pv", "bs", "npv", "ue"],
+            geobox=task.geobox,
+            native_transform=self._native_tr,
+            fuser=_nodata_fuser,
+            groupby="solar_day",
+            resampling=self.resampling,
+            chunks=chunks,
+        )
+
+        xx["wet"] = load_with_native_transform(
+            task.datasets,
+            bands=["water"],
+            geobox=task.geobox,
+            native_transform=self._native_tr_wet,
+            fuser=_or_fuser,
+            groupby="solar_day",
+            resampling=self.resampling,
+            chunks=chunks,
+        )
```

Comment:
This will probably result in loading `water` band twice.

To avoid that you need to return `wet` from `input_data`, then you'll need to implement custom fuser that will fuse data bands differently from the `wet` band.

Reply:
yeah it does unfortunately, I could do something like make a custom fuser that calls the `_or_fuser` if a boolean type is detected and calls `_nodata_fuser` otherwise

Reply:
no need to "detect", you know exactly which bands to expect on input and which bands need to be fused with what logic.